xonsh/cli_utils.py

```diff
@@ -51,58 +94,289 @@ def get_doc(func: tp.Callable, parameter: str = None):
     str
         doc of the parameter/function
     """
+    if isinstance(func, str):
+        return func
     import inspect
 
     doc = inspect.getdoc(func) or ""
     if parameter:
-        return _get_param_doc(doc, parameter)
+        return inspect.cleandoc(_get_param_doc(doc, parameter)).strip()
     else:
-        return _get_func_doc(doc)
+        return _get_func_doc(doc).strip()
 
 
 _FUNC_NAME = "_func_"
 
 
+def _get_args_kwargs(annot: tp.Any) -> tp.Tuple[tp.Sequence[str], tp.Dict[str, tp.Any]]:
+    args, kwargs = [], {}
+    if isinstance(annot, tuple):
+        args, kwargs = annot
+    elif TYPING_ANNOTATED_AVAILABLE and "Annotated[" in str(annot):
```

Comment:
Isn't there a better way than checking the string?

Reply:
the other way involves more code and digging some implementation detail. the generics changed in py3.7 from how it used to have __origin__ attrs ...